src/nimib.nim

```diff
@@ -189,14 +189,27 @@ when moduleAvailable(karax/kbase):
     nbRawHtml: "<div id=\"" & rootId & "\"></div>"
     nbKaraxCodeBackend(rootId, args)
 
-template nbJsShowSource*(message: string = "") =
+template nbJsShowSource*(message: string = "") {.deprecated: "Use nbCodeDisplay instead".} =
   nb.blk.context["js_show_nim_source"] = true
   if message.len > 0:
     nb.blk.context["js_show_nim_source_message"] = message
 
-template nbCodeToJsShowSource*(message: string = "") {.deprecated: "Use nbJsShowSource instead".} =
+template nbCodeToJsShowSource*(message: string = "") {.deprecated: "Use nbCodeDisplay instead".} =
   nbJsShowSource(message)
 
+template nbCodeDisplay*(tmplCall: untyped, body: untyped) =
+  ## display codes used in a template (e.g. nbJsFromCode) after the template call
+  tmplCall:
+    body
+  newNbCodeBlock("nbCode", body):
+    discard
+
+template nbCodeAnd*(tmplCall: untyped, body: untyped) =
+  ## can be used to run code both in c and js backends (e.g. nbCodeAnd(nbJsFromCode))
+  nbCode: # this should work because template name starts with nbCode
```

Comment:
It's nice that we are embracing this lucky bug at least :rofl:

Reply:
yeah, maybe too much... still hopeful that at some point we will have a better solution for a composable CodeAsInSource (not working on it, just hope 😄 )

Reply:
Yeah, I'll try and think about it from time to time to see if I can get any strokes of genius. It's starting to bug me that it doesn't compose well. There's too much jumping through hoops to reuse a template 😅 So this might be my next issue to solve...